cmd/cmd.go

```diff
@@ -66,12 +76,12 @@ func CommandLine(driver, targetDb, projectID, instanceID, dbName string, dataOnl
 		}
 	}
 
-	dbURI, err := conversion.CreateOrUpdateDatabase(projectID, instanceID, dbName, dbExists, conv, ioHelper.Out)
+	err = conversion.CreateOrUpdateDatabase(ctx, adminClient, projectID, instanceID, dbName, dbExists, conv, ioHelper.Out)
```

Comment:
We should also clean up passing of projectID, instanceID and dbName as part of this cleanup by precomputing dbURI at the top level once instance flag is validated and passing the same around.

Reply:
I cleaned up project and instance everywhere except this function and the ones this calls because this eventually calls analyzeError() which takes the instance and project and input to print a clean error message. I was wondering what we should do with that if we decide to remove that from here?

Reply:
As discussed in other comment and offline, we can use the splitDbURI (or parseDbURI) in analyzeError() to split passed in dbURI into component projectId, instanceId and dbName etc.